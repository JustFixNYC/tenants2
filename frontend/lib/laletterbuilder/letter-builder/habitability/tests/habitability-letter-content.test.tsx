import React from "react";
import ReactTestingLibraryPal from "../../../../tests/rtl-pal";
import {
  HabitabilityLetterContent,
  habitabilitySampleLetterProps,
  getHabitabilityLetterContentPropsFromSession,
} from "../habitability-letter-content";
import { newSb } from "../../../../tests/session-builder";

describe("<HabitabilityContent>", () => {
  it("renders", () => {
    const pal = new ReactTestingLibraryPal(
      (
        <HabitabilityLetterContent
          {...habitabilitySampleLetterProps}
          todaysDate="2020-06-10"
        />
      )
    );
    expect(pal.rr.container).toMatchSnapshot();
  });

  it("includes issues and rooms", () => {
    const pal = new ReactTestingLibraryPal(
      (
        <HabitabilityLetterContent
          {...habitabilitySampleLetterProps}
          todaysDate="2020-06-10"
        />
      )
    );
    expect(pal.rr.getAllByText("{roomLabel}")).toHaveLength(2);
    expect(pal.rr.getAllByText("{issueLabel}")).toHaveLength(1);
  });

  it("translates into spanish", () => {});
});

const filledUser = newSb().withLoggedInJustfixUser().withLandlordDetails();

describe("getHabitabilityContentPropsFromSession()", () => {
  it("returns null when user is logged out", () => {
    expect(getHabitabilityLetterContentPropsFromSession(newSb().value)).toBe(
      null
    );
  });

  it("returns expected value when user is logged in", () => {
    expect(
      getHabitabilityLetterContentPropsFromSession(filledUser.value)
    ).toMatchSnapshot();
  });
});
