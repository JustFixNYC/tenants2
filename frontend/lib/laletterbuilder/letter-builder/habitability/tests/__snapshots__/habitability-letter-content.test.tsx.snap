// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitabilityContent> works 1`] = `
<div>
  <h1
    class="has-text-right"
    style="white-space: pre-wrap;"
  >
    Habitability
  </h1>
  <div
    class="jf-page-break-after"
  >
    <p
      class="has-text-right"
    >
      Wednesday, June 10, 2020
    </p>
    <dl
      class="jf-letter-heading"
    >
      <dt>
        To
      </dt>
      <dd>
        LANDLORDO CALRISSIAN
        <br />
        1 Cloud City Drive
        <br />
        Bespin, OH 41235
      </dd>
      <dt>
        From
      </dt>
      <dd>
        Boop Jones
        <br />
        654 Park Place #2
        <br />
        Brooklyn
        , 
        NY
         
        12345
        <br />
        (555) 123-4567
      </dd>
    </dl>
    <p>
      Dear 
      LANDLORDO CALRISSIAN
      ,
    </p>
    <p>
      laletterbuilder.habitability.intro
    </p>
    <p
      class="jf-signature"
    >
      Signed,
      <br />
      <br />
      Boop Jones
    </p>
  </div>
</div>
`;

exports[`getHabitabilityContentPropsFromSession() returns expected value when user is logged in 1`] = `
Object {
  "aptNumber": "",
  "city": "Brooklyn",
  "email": "boop@jones.net",
  "firstName": "Boop",
  "landlordAddress": "123 Cloud City Drive
Bespin, NY 12345",
  "landlordEmail": "landlordo@calrissian.net",
  "landlordName": "Landlordo Calrissian",
  "lastName": "Jones",
  "phoneNumber": "5551234567",
  "state": "NY",
  "street": "150 Court St",
  "zipCode": "11201",
}
`;
